### DISCUSSION TITLE
Interactive graph

### DISCUSSION THREAD

**Discussion by bart-krakowski** [mermaid-js/mermaid#6122] [Ideas]
Exploring huge graphs of dependencies/relationships etc. is hard. Would it be possible to implement something similar to what NX does, where you hover over a specific node and it highlight the related ones?

https://nx.dev/features/explore-graph

**Comment by Stanton** [comment]
You can achieve this if you're willing to figure out the associations yourself, I maintain a json model of my data structure outside of Mermaid and query that as my users hover over nodes, letting me highlight the paths between and through specific nodes by targeting the IDs that mermaid uses with JS and CSS.

![interactions](https://github.com/user-attachments/assets/6b0920b7-3f21-48a7-93bc-504ed9262f99)

That being said, a more native framework to achieve interactions would be gratefully received!
